File: java/ecbenchmark/src/ecbenchmark/operator/Grouped.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ecbenchmark.operator;

import ecbenchmark.Operator;
import java.util.Arrays;
import ecbenchmark.Function;

/**
 *
 * @author jcrada
 */
public class Grouped extends Operator {

    private int fromIndex, toIndex;

    public Grouped(int from, int to, Function f) {
        super(f);
        this.fromIndex = from;
        this.toIndex = to;
    }

    @Override
    public double f(final double[] x) {
        double[] group = Arrays.copyOfRange(x, fromIndex, toIndex);
        return innerFunction.f(group);
    }
}

File: cpp/ecbenchmark/src/operator/Grouped.cpp
#include "ecbenchmark/operator/Grouped.h"

namespace ecb {

    Grouped::Grouped(int from, int to, Function* f)
    : Operator(f), _fromIndex(from), _toIndex(to) {

    }

    void Grouped::setFromIndex(int from) {
        this->_fromIndex = from;
    }

    int Grouped::fromIndex() const {
        return this->_fromIndex;
    }

    void Grouped::setToIndex(int to) {
        this->_toIndex = to;
    }

    int Grouped::toIndex() const {
        return this->_toIndex;
    }

    scalar Grouped::f(const std::vector<scalar>& x) {
        std::vector<scalar> grouped_x(x.begin() + _fromIndex, x.begin() + _toIndex);
        return _innerFunction->f(grouped_x);
    }

}

